### PR TITLE
feat(): implement content block nsfw handler

### PR DESCRIPTION
**Pull request by didd**
## Description

- add eighteen plus icon placeholder
- implement content block nsfw handler

## Checklist

- [X] I have read the **README** document
- [X] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [X] All new and existing tests passed
- [X] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx [hunk 1]
```diff
@@ -72,6 +75,17 @@ export const BeamEditor: React.FC = () => {
 
   const handleNsfwCheckbox = () => {
     setIsNsfw(!isNsfw);
+
+    if (!isNsfw) {
+      range(0, blocksInUse.length).forEach(key => {
+        setNsfwBlocks(new Map(nsfwBlocks.set(key, true)));
```
Comment (kenshyx) [inline ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: Can you please explain the logic in this loop?

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: It sets all content blocks on the editor as nsfw blocks when the nsfw checkbox on the header(beam) is set. 
image

Comment (kenshyx) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: Why use `range` function from lodash instead of a normal for loop?

Comment (kenshyx) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: also is there a need to create n*Map objects where `n` is the number of blocks? can you please explain the logic?

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: It's meant to make it declarative and read better but I see from your comment now that may not be  what we have here. I will revert to the normal for loop.

Comment (kenshyx) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: There are also fundamental differences between a forEach function and a for loop. You can never exit from a forEach.

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: Indeed

Comment (kenshyx) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: also the overhead of that range function from lodash is crazy https://github.com/lodash/lodash/blob/main/src/.internal/baseRange.ts

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: It's not worth it. Initially I considered the overhead as well but gave weight to readability.

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: > also is there a need to create n*Map objects where n is the number of blocks? can you please explain the logic?

It's to make react aware that I have a state change and for it to cause a re-render. That's why I am doing this :

`setNsfwBlocks(new Map(nsfwBlocks.set(key, true)))`

Comment (kenshyx) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: ok but couldn't this be done in one update using just 1 Map object?

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: You are right. We can do that. :D That one map object is enough indeed.

Comment (didd) [reply ui/apps/akasha/src/extensions/beam-editor/beam-editor.tsx]: Thanks!

**Review by kenshyx — APPROVED** [review]

**Pull request merged**